modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java

```diff
@@ -1604,6 +1616,13 @@ public void onLocalJoin(DiscoveryEvent evt, DiscoCache discoCache) {
             // First node start, method onGridDataReceived(DiscoveryDataBag.GridDiscoveryData) has not been called.
             ArrayList<ServiceInfo> staticServicesInfo = staticallyConfiguredServices(false);
 
+            if (ctx.security().enabled()) {
```

Comment:
Security enabled check can be placed in the `checkDeployPermissionDuringJoin` method.